drivers/comparator/comparator_stm32_comp.c

```diff
@@ -0,0 +1,321 @@
+		callback(dev, user_data);
+	}
+
+	return 0;
+}
+
+static DEVICE_API(comparator, stm32_comp_comp_api) = {
+	.get_output = stm32_comp_get_output,
+	.set_trigger = stm32_comp_set_trigger,
+	.set_trigger_callback = stm32_comp_set_trigger_callback,
+	.trigger_is_pending = stm32_comp_trigger_is_pending,
+};
+
+static int stm32_comp_pm_callback(const struct device *dev, enum pm_device_action action)
+{
+	const struct stm32_comp_config *cfg = dev->config;
+	COMP_TypeDef *comp = cfg->comp;
+
+	if (action == PM_DEVICE_ACTION_RESUME) {
+		if (cfg->lock_enable) {
+			LL_COMP_Lock(comp);
+		}
+		LL_COMP_Enable(comp);
+	}
+
+	if (IS_ENABLED(CONFIG_PM_DEVICE) && action == PM_DEVICE_ACTION_SUSPEND) {
+		LL_COMP_Disable(comp);
+	}
+
+	return 0;
+}
+
+static void stm32_comp_irq_handler(const struct device *dev)
+{
+	const struct stm32_comp_config *cfg = dev->config;
+	struct stm32_comp_data *data = dev->data;
+
+	if (stm32_exti_is_pending(cfg->exti_line_number)) {
+		stm32_exti_clear_pending(cfg->exti_line_number);
+	}
+
+	if (data->callback == NULL) {
+		return;
+	}
+
+	data->callback(dev, data->user_data);
+}
+
+static int stm32_comp_init(const struct device *dev)
+{
+	const struct device *const clk = DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE);
+	const struct stm32_comp_config *cfg = dev->config;
+	COMP_TypeDef *comp = cfg->comp;
+	int ret = 0;
+
+	if (!device_is_ready(clk)) {
+		LOG_ERR("%s clock control device not ready", dev->name);
+		return -ENODEV;
+	}
+
+	/* Enable COMP bus clock */
+	ret = clock_control_on(clk, &cfg->pclken[0]);
+	if (ret != 0) {
+		LOG_ERR("%s clock op failed (%d)", dev->name, ret);
+		return ret;
+	}
+
+	/* Enable COMP clock source */
+	clock_control_configure(clk, &cfg->pclken[1], NULL);
+	if (ret != 0) {
+		LOG_ERR("%s clock configure failed (%d)", dev->name, ret);
+		return ret;
+	}
+
+	/* Configure COMP inputs as specified in Device Tree, if any */
+	ret = pinctrl_apply_state(cfg->pincfg, PINCTRL_STATE_DEFAULT);
+	if (ret < 0 && ret != -ENOENT) {
+		/*
+		 * If the COMP is used only with internal channels, then no pinctrl is
+		 * provided in Device Tree, and pinctrl_apply_state returns -ENOENT,
+		 * but this should not be treated as an error.
+		 */
+		LOG_ERR("%s pinctrl setup failed (%d)", dev->name, ret);
+		return ret;
+	}
+
+	if (LL_COMP_IsLocked(comp)) {
+		/* COMP instance shall not be locked */
+		LOG_ERR("%s COMP instance is locked", dev->name);
+		return -EACCES;
+	}
+
+#ifndef CONFIG_SOC_SERIES_STM32G4X
+	LL_COMP_SetPowerMode(comp, cfg->power_mode);
+#endif /* !CONFIG_SOC_SERIES_STM32G4X */
+
+	LL_COMP_SetInputMinus(comp, cfg->input_minus);
+	LL_COMP_SetInputPlus(comp, cfg->input_plus);
+	LL_COMP_SetInputHysteresis(comp, cfg->hysteresis);
+	LL_COMP_SetOutputPolarity(comp, cfg->invert_output);
+	LL_COMP_SetOutputBlankingSource(comp, cfg->blank_sel);
+
+	if (cfg->miller_effect_hold_enable) {
+#if defined(CONFIG_SOC_SERIES_STM32G4X)
+		SET_BIT(comp->CSR, BIT(1U));
```

Comment:
I could not find this bit described in the reference manual. Do you have a pointer?

Reply:
This is maybe already 6th time I am referencing to this URL: https://community.st.com/t5/stm32-mcus-products/stm32g431-comparator-bug/m-p/679540#M250779
The url is also added to .yml file and leads to ST's forum. The feature is not documented, but solves real problems. Please take a look on it 😄

Reply:
> This is maybe already 6th time I am referencing to this URL: https://community.st.com/t5/stm32-mcus-products/stm32g431-comparator-bug/m-p/679540#M250779

Hahaha, my apologies, I should have checked the bindings documentation.
Indeed, an inline comment here would help, e.g. `/* Undocumented bit, refer to st,miller-effect-hold-enable DT binding */`.

Reply:
done